app/controllers/chat_controller.rb

```diff
@@ -149,6 +149,9 @@ def update_user_last_read
 
     membership = UserChatChannelMembership.find_by(user: current_user, chat_channel: @chat_channel, following: true)
     raise Discourse::NotFound if membership.nil?
+    chat_message = ChatMessage.find_by(chat_channel_id: @chat_channel.id, id: params[:message_id])
+    raise Discourse::NotFound unless chat_message
+
```

Comment:
We should probably check that `params[:message_id]`is greater than `membership#last_read_message_id` to prevent the frontend from accidentally "unreading" messages. What do you think?

Reply:
ok :+1:

Reply:
However we should do it in the future IMO otherwise it might conflict with resolving the problem here as it might always consider the last_read_message_id is greater given it's an incorrect one

Reply:
For the record (because I was confused 😉 ).

It looks like this check was added, and the problem with existing channels has been handled via the post-deploy migration :+1: Marking as resolved